experimental/generation/generator/src/library/dialogGenerator.ts

```diff
@@ -644,7 +662,7 @@ export async function generate(
             await fs.emptyDir(outPathSingle)
         }
 
-        let standard = normalize(ppath.join(__dirname, '../../templates'))
+        let standard = normalize(ppath.join(__dirname, '../../templates/standard'))
         templateDirs = expandStandard(templateDirs)
```

Comment:
We can drop expandStandard.  They way it should work:
1) We join specified template dirs + standard and professional and use these to find schemas.  This makes it easy to refer to standard.schema, professional.schema or other stuff we add.  
2) For looking for templates we use template dirs in the order specified plus any dirs with schema in the order was used. (That is what the code below after finding schema does.)

Reply:
All I need to do is just deleting the expandStandard function ?

Reply:
Delete the function and this call.  You also need to ensure all our standard templates are included when we process schemas, i.e. you want the below so you can find all schemas we define.
<passedIn templates>, templates/standard, templates/professional.  (Bonus point for just enumerating templates directory to generate this.)  
Once you processSchemas, you change to <passedIn templates> + <schema template directories>, i.e. what is passed in plus template directories associated with schemas people use.

---
In reply to: [436688418](https://github.com/microsoft/BotBuilder-Samples/pull/2474#discussion_r436688418) [](ancestors = 436688418)